app/src/main/java/com/example/tasky/feature_agenda/domain/model/AgendaResponse.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.tasky.feature_agenda.domain.model
+
+import com.example.tasky.agenda_details.domain.model.EventResponse
+import com.example.tasky.agenda_details.domain.model.ReminderResponse
+import com.example.tasky.agenda_details.domain.model.TaskResponse
+
+data class AgendaResponse(
+    val events: List<EventResponse> = emptyList(),
+    val tasks: List<TaskResponse> = emptyList(),
+    val reminders: List<ReminderResponse> = emptyList()
+)
+
+data class SyncAgendaResponse(
+    val deletedEventIds: List<String> = emptyList(),
+    val deletedTasksIds: List<String> = emptyList(),
+    val deletedReminderIds: List<String> = emptyList()
+)
```

Comment:
For this response (if I'm not wrong it should actually be a request, right?) it's similar, since in the end outside of the data layer, there's no use of knowing which items were deleted locally and have to be synced. If there's an agenda item that you deleted locally and still have to sync, you'll have to keep a persistent reference to it anyways in your DB. And if you have that reference in your DB, you can just get it in your data layer as well without having to pass this information through all of the layers. Here some pseudo code what I mean:
```
val deletedEventIds = deletedEventDao.getAllLocallyDeletedEventIds()
val deletedTaskIds = ...
val deletedReminderIds = ...

// This object is now created in data directly, so you don't need this information in domain anymore
val syncRequest = SyncAgendaRequest(
    deletedEventIds = deletedEventIds,
    // ...
)
```